--- Makefile.in	2010-05-01 01:31:10.568544865 +0200
+++ Makefile.in.new	2010-05-01 02:01:00.145031894 +0200
@@ -505,7 +505,7 @@
 
 install-arch:
 	@# Arch dependent files
-	@find system screensavers visualisations -regextype posix-extended -type f -not -iregex ".*svn.*|.*win32(dx)?\.vis|.*osx\.vis" -iregex ".*$(ARCH).*|.*\.vis|.*\.xbs" -exec install -D "{}" $(prefix)/share/xbmc/"{}" \; -printf " -- %-75.75f\r"	
+	@find system -regextype posix-extended -type f -not -iregex ".*svn.*|.*win32(dx)?\.vis|.*osx\.vis" -iregex ".*$(ARCH).*|.*\.vis|.*\.xbs" -exec install -D "{}" $(prefix)/share/xbmc/"{}" \; -printf " -- %-75.75f\r"	
 
 install-scripts:
 	@install -D tools/Linux/xbmc.sh $(prefix)/bin/xbmc
