--- Makefile.in.orig	2010-04-29 14:54:21.000000000 +0200
+++ Makefile.in	2010-04-29 16:27:55.000000000 +0200
@@ -521,8 +521,6 @@
 	@echo "Copying system files to $(prefix)/share/xbmc"
 	@# Arch independent files
 	@find addons language media scripts sounds userdata system -regextype posix-extended -type f -not -iregex ".*svn.*|.*\.so|.*\.dll|.*\.pyd|.*python/.*\.zlib" -exec install -D -m 0644 "{}" $(prefix)/share/xbmc/"{}" \; -printf " -- %-75.75f\r"
-	@# Skins
-	@find skin -regextype posix-extended -type f -not -iregex '.*/\.svn/.*|^skin/[^/]*/media/.*[^x][^b][^t]' -exec install -D -m 0644 '{}' $(prefix)/share/xbmc/'{}' \; -printf " -- %-75.75f\r"
 	@# Icons and links
 	@mkdir -p $(prefix)/share/applications $(prefix)/share/pixmaps
 	@cp -a tools/Linux/xbmc.png $(prefix)/share/pixmaps/
