Index: xbmc/visualizations/XBMCProjectM/libprojectM/PresetLoader.cpp
===================================================================
--- xbmc/visualizations/XBMCProjectM/libprojectM/PresetLoader.cpp	(revisione 26936)
+++ xbmc/visualizations/XBMCProjectM/libprojectM/PresetLoader.cpp	(copia locale)
@@ -77,11 +77,19 @@
     return; // no files loaded. m_entries is empty
   }
 
-  struct dirent * dir_entry;
+  #ifdef LINUX
+    struct dirent64 * dir_entry;
+  #else
+    struct dirent * dir_entry;
+  #endif
   std::set<std::string> alphaSortedFileSet;
   std::set<std::string> alphaSortedPresetNameSet;
   
-  while ((dir_entry = readdir(m_dir)) != NULL)
+  #ifdef LINUX
+    while ((dir_entry = readdir64(m_dir)) != NULL)
+  #else
+    while ((dir_entry = readdir(m_dir)) != NULL)
+  #endif
   { 
 
     std::ostringstream out;
