diff -Nru a/backends/pacman/backend-pacman.c b/backends/pacman/backend-pacman.c
--- a/backends/pacman/backend-pacman.c	2010-11-30 00:40:11.000000000 +1300
+++ b/backends/pacman/backend-pacman.c	2010-12-22 13:42:31.100000350 +1300
@@ -254,6 +254,7 @@
 	backend_simulate_install_packages,	/* simulate_install_packages */
 	backend_simulate_remove_packages,	/* simulate_remove_packages */
 	backend_simulate_update_packages,	/* simulate_update_packages */
+	NULL,					/* upgrade_packages */
 	NULL,					/* transaction_start */
 	NULL					/* transaction_stop */
 );
