diff -ru a/kwin/qtcurveclient.cpp b/kwin/qtcurveclient.cpp
--- a/kwin/qtcurveclient.cpp	2012-04-24 17:20:01.000000000 -0400
+++ b/kwin/qtcurveclient.cpp	2012-06-09 23:52:15.049850728 -0400
@@ -852,7 +852,7 @@
     bool showIcon=TITLEBAR_ICON_NEXT_TO_TITLE==Handler()->wStyle()->pixelMetric((QStyle::PixelMetric)QtC_TitleBarIcon,  0L, 0L);
     int  iconSize=showIcon ? Handler()->wStyle()->pixelMetric(QStyle::PM_SmallIconSize) : 0;
 
-#if KDE_IS_VERSION(4, 3, 85)
+#if KDE_IS_VERSION(4, 3, 85) && !KDE_IS_VERSION(4, 8, 80)
     QList<ClientGroupItem> tabList  = clientGroupItems();
     const int              tabCount = tabList.count();
 
@@ -922,7 +922,7 @@
                                                    rectX2-(titleEdgeRight+rectX+titleEdgeLeft),
                                                    itsCaptionRect.height()),
                    itsCaption, showIcon ? icon().pixmap(iconSize) : QPixmap(), shadowSize);
-#if KDE_IS_VERSION(4, 3, 85)
+#if KDE_IS_VERSION(4, 3, 85) && !KDE_IS_VERSION(4, 8, 80)
 
         if(constAddToEmpty==itsTargetTab)
         {
@@ -947,7 +947,7 @@
     //     if(itsHover)
         {
             if(
-#if KDE_IS_VERSION(4, 3, 85)
+#if KDE_IS_VERSION(4, 3, 85) && !KDE_IS_VERSION(4, 8, 80)
                 1==tabCount &&
 #endif
                 active && (itsToggleMenuBarButton||itsToggleStatusBarButton))
@@ -1322,7 +1322,7 @@
 //         return true;
 //     }
     
-#if KDE_IS_VERSION(4, 3, 85)
+#if KDE_IS_VERSION(4, 3, 85) && !KDE_IS_VERSION(4, 8, 80)
     if(Handler()->grouping())
     {
         if(QtCurveButton *btn = dynamic_cast<QtCurveButton *>(o))
@@ -1360,7 +1360,7 @@
     return KCommonDecoration::eventFilter(o, e);
 }
 
-#if KDE_IS_VERSION(4, 3, 85)
+#if KDE_IS_VERSION(4, 3, 85) && !KDE_IS_VERSION(4, 8, 80)
 bool QtCurveClient::mouseButtonPressEvent(QMouseEvent *e)
 {
     itsClickPoint = widget()->mapToParent(e->pos());
diff -ru a/kwin/qtcurvehandler.cpp b/kwin/qtcurvehandler.cpp
--- a/kwin/qtcurvehandler.cpp	2012-04-24 17:20:01.000000000 -0400
+++ b/kwin/qtcurvehandler.cpp	2012-06-09 23:48:56.669671970 -0400
@@ -254,7 +254,7 @@
         case AbilityProvidesShadow:
             return customShadows();
 #endif
-#if KDE_IS_VERSION(4, 3, 85)
+#if KDE_IS_VERSION(4, 3, 85) && !KDE_IS_VERSION(4, 8, 80)
         case AbilityClientGrouping:
             return grouping();
 #endif
