# HG changeset patch
# Parent 296ea099157e159eff3a7e98d85776ece38980d4
diff --git a/src/if_ruby.c b/src/if_ruby.c
--- a/src/if_ruby.c
+++ b/src/if_ruby.c
@@ -335,12 +335,12 @@
 static void* (*ruby_process_options)(int, char**);
 #endif
 
-#ifdef RUBY19_OR_LATER
-static SIGNED_VALUE rb_num2long_stub(VALUE x)
+#if defined(RUBY19_OR_LATER) && !defined(PROTO)
+SIGNED_VALUE rb_num2long_stub(VALUE x)
 {
     return dll_rb_num2long(x);
 }
-static VALUE rb_int2big_stub(SIGNED_VALUE x)
+VALUE rb_int2big_stub(SIGNED_VALUE x)
 {
     return dll_rb_int2big(x);
 }
