--- a/vboxnetflt/linux/VBoxNetFlt-linux.c
+++ b/vboxnetflt/linux/VBoxNetFlt-linux.c
@@ -1805,5 +1805,9 @@
 {
     PVBOXNETFLTINS      pThis = VBOX_FLT_NB_TO_INST(self);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)
+    struct net_device  *pDev  = netdev_notifier_info_to_dev(ptr);
+#else
     struct net_device  *pDev  = (struct net_device *)ptr;
+#endif
     int                 rc    = NOTIFY_OK;
 
