--- vboxvideo/vboxvideo_drm.c.org	2014-10-16 12:36:14.000000000 +0200
+++ vboxvideo/vboxvideo_drm.c	2014-11-10 01:49:15.353329725 +0100
@@ -77,6 +77,7 @@
 # endif
 
 #include "drm/drmP.h"
+#include "drm/drm_legacy.h"
 #include "vboxvideo_drm.h"
 
 # ifndef RHEL_RELEASE_CODE
@@ -122,7 +123,7 @@
         .open = drm_open,
         .release = drm_release,
         .unlocked_ioctl = drm_ioctl,
-        .mmap = drm_mmap,
+        .mmap = drm_legacy_mmap,
         .poll = drm_poll,
 };
 #endif
@@ -152,7 +153,7 @@
 #else
         .ioctl = drm_ioctl,
 #endif
-        .mmap = drm_mmap,
+        .mmap = drm_legacy_mmap,
         .poll = drm_poll,
     },
 #else /* LINUX_VERSION_CODE >= KERNEL_VERSION(3, 3, 0) || defined(DRM_FOPS_AS_POINTER) */
